src/components/theme/useLang.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+import {useSyncExternalStore} from 'use-sync-external-store/shim';
+
+import type {StringWithSuggest} from '../../types/utils';
+import type {Lang} from '../../utils/configure';
+import {getConfig, subscribeConfigure} from '../../utils/configure';
+
+export interface LangOptions {
+    lang: StringWithSuggest<Lang>;
+    fallbackLang: StringWithSuggest<Lang>;
+}
+
+export const defaultLangOptions: LangOptions = {lang: 'en', fallbackLang: 'en'};
+
+export const LangContext = React.createContext<LangOptions | undefined>(undefined);
+
+export function useLang(): LangOptions {
+    const config = useSyncExternalStore(
+        (onChange) => subscribeConfigure(onChange),
```

Comment:
React docs recommend to memoize callbacks for this hook, i'm not sure if same applies to this package

Reply:
I was reffering to this

> If a different subscribe function is passed during a re-render, React will re-subscribe to the store using the newly passed subscribe function. You can prevent this by declaring subscribe outside the component.